packages/client-ts/src/client.ts

```diff
@@ -203,9 +203,29 @@ class Client {
   public async openPublicProfile(
     did: string,
     contextName: string,
-    profileName: string = "basicProfile"
+    profileName: string = "basicProfile",
+    fallbackContext: string = "Verida: Vault"
   ): Promise<Profile | undefined> {
-    const context = await this.openExternalContext(contextName, did);
+    const maxRetry = 2;
```

Comment:
An alternative implementation here could avoid using a retry count.

Basically, when the error is caught (L220), if `fallbackContext` isn't empty or null, return this.openPublicProfile(...) and set `fallbackContext` to null. Next time the error is caught, it will be rethrown because there is no remaining `fallbackContext`.

Reply:
You're right. Thanks @tahpot

Reply:
updated, funny how I wrote so much verbose code 🥲 

@tahpot should I merge this PR right away or wait for later.